contracts/FlowIDTableStaking.cdc

```diff
@@ -422,8 +422,12 @@ pub contract FlowIDTableStaking {
         /// which means that it is a new node who currently is not participating with tokens staked
         /// and has enough committed for the next epoch for its role
         access(self) fun isEligibleForCandidateNodeStatus(_ nodeRecord: &FlowIDTableStaking.NodeRecord): Bool {
+            let participantList = FlowIDTableStaking.getParticipantNodeList()!
```

Comment:
This would account for the case where an operator adds tokens to an existing participant node. 

What if an approved node un-stakes then re-stakes?

Reply:
If they un-stake and then re-stake in the same epoch, then they're still in the participant list, so this would still return false because they are still in the participant list. 

If they un-stake below the minimum in one epoch and then re-stake above the minimum in a future epoch, they will have been removed from the participant list, so this will return true because they are no longer a participant and can be considered a candidate node.

Does that answer your question?

Reply:
> If they un-stake below the minimum in one epoch and then re-stake above the minimum in a future epoch, they will have been removed from the participant list, so this will return true because they are no longer a participant and can be considered a candidate node.

If an Access Node on the allow-list went through this process (un-stake then re-stake in a future epoch), wouldn't we have the same problem?
- The AN would be considered a candidate node (`isEligibleForCandidateNodeStatus == true`)
- The AN would then be subject to slot selection
- If the AN does not win a slot, they would be removed and refunded but still be included in `getProposedNodeIDs`

Reply:
A few days ago on Slack we were discussing having `removeAndRefundNode` also remove a node from the allow-list. I think incorporating that change here would solve this problem.

Reply:
you're right. I'll update `removeAndRefundNode` to remove it from the approve list